src/hooks/useKeyboardShortcut/index.ts

```diff
@@ -0,0 +1,74 @@
+import { useCallback, useEffect, useRef } from "react";
+
+type UseKeyboardShortcutOptions = {
+  preventDefault?: boolean;
+  overrideIgnore?: boolean;
+};
+
+// Used to prevent shortcuts from being activated on input elements.
+const INPUT_ELEMENTS = ["INPUT", "TEXTAREA", "SELECT"];
+
+const useKeyboardShortcut = (
+  shortcutKeys: string[],
+  cb: () => void,
+  disabled: boolean = false,
+  options: UseKeyboardShortcutOptions = {}
+) => {
+  // We wrap the callback to prevent triggering unnecessary useEffect.
+  const cbRef = useRef(cb);
+  cbRef.current = cb;
+
+  const isShortcutPressed = useCallback(
+    (event: KeyboardEvent) => {
+      if (
+        shortcutKeys.includes("Control") &&
+        !event.ctrlKey &&
+        !event.metaKey
+      ) {
+        return false;
+      }
+      if (shortcutKeys.includes("Alt") && !event.altKey) return false;
+      if (shortcutKeys.includes("Shift") && !event.shiftKey) return false;
+
+      return shortcutKeys[shortcutKeys.length - 1] === event.key;
+    },
+    [shortcutKeys]
+  );
```

Comment:
The behavior of this isn't consistent if a user doesn't pass in a special key.  For example if i want to trigger this hook off of only one key press i can no longer use an input with that character.

In the below example i set a keyboard shortcut for the "s" and "f" keys i can no longer use the text input and type those characters.

https://user-images.githubusercontent.com/4605522/195444299-e3ad79b7-3cd9-46ef-859a-a1336436f8c2.mov

An example when this may come up if we introduce a keyboard shortcut to allow a user to paginate  across their search results e.g. `press "n" to paginate forward or "p" for back`